### ISSUE TITLE
Inclure les autres tables dvf

### ISSUE THREAD

**Issue by soufianekhoudmi** [make-open-data/make-open-data#41]
En tant qu'analyste, je veux l'ensemble des tables dvf, actuellement seulement 2023 est incluse

La DGFIP fournit dvf pour les 5 dernière années, actuellement seule la dernière est utilisée.

- Ajouter les autres sources dans extract sources.yml
- Etendre le modele immo/grouper_mutation.sql pour traiter les 5 années puis les fuisinner

**Comment by soufianekhoudmi** [comment]
Lien toutes les années dvf https://files.opendatarchives.fr/cadastre.data.gouv.fr/data/etalab-dvf/latest/csv/

**Comment by aloysius-tim** [comment]
Et la version brute, non géolocalisée  -> https://data.cquest.org/dgfip_dvf/

Je pense que le traitement appliquée sur la version géolocalisée est le suivant: https://github.com/datagouv/dvf?tab=readme-ov-file

**Comment by soufianekhoudmi** [comment]
> Et la version brute, non géolocalisée -> https://data.cquest.org/dgfip_dvf/
> 
> Je pense que le traitement appliquée sur la version géolocalisée est le suivant: https://github.com/datagouv/dvf?tab=readme-ov-file

Pas sûr qu'il soit possible de géolocaliser avec dbt dans le cadre de MOD. 
Mieux vaut utiliser les versions géolocalisées.

**Comment by soufianekhoudmi** [comment]
Salut Tim, 

Je ne trouve pas les dvf les plus récentes sur le lien partagé. 

<img width="870" alt="Capture d’écran 2024-09-30 à 10 11 53" src="https://github.com/user-attachments/assets/61b216f7-b4c6-48ab-b711-99534edd18dd">

Je viens d'uploader les csv ici https://www.data.gouv.fr/fr/datasets/demandes-de-valeurs-foncieres-geolocalisees/ dans storage. 
- dvf_2023 : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2023/full.csv (utilisé jusqu'à présent)
- dvf_2022 : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2022/full.csv
- dvf_2021 : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2021/full.csv
- dvf_2020 : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2020/full.csv
- dvf_2019 : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2019/full.csv

Je viens également d'uploader des csv plus réduit à utiliser pour tester (target autre que production)
- dvf_2023_dev : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2023_dev/dvf_2023_dev.csv (utilisé jusqu'à présent)
- dvf_2022_dev : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2022_dev/34.csv
- dvf_2021_dev : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2021_dev/34.csv
- dvf_2020_dev : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2020_dev/34.csv
- dvf_2019_dev : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2019_dev/34.csv

**Comment by aloysius-tim** [comment]
Hello Soufiane,
Ah top ! Je suis justement en train de bosser sur l'integration de tous les millésimes en ce moment !
En effet, pour les plus récentes, elles sont dispos sur le repo officiel. Donc mix du repo officiel pour > 2019 et openarchives.fr pour < 2019

**Comment by aloysius-tim** [comment]
D'ailleurs, une raison particulière de filtrer sur APPART/MAISON en ACHAT ? 

```
--      WHERE 
--         EXISTS (
--             SELECT 1
--             FROM source_dvf d1
--             WHERE d1.id_mutation = source_dvf.id_mutation AND d1.type_local IN ('Appartement', 'Maison')
--         ) AND
--         NOT EXISTS (
--             SELECT 1
--             FROM source_dvf d2
--             WHERE d2.id_mutation = source_dvf.id_mutation AND d2.nature_mutation != 'Vente'
--         )
```

**Comment by soufianekhoudmi** [comment]
> D'ailleurs, une raison particulière de filtrer sur APPART/MAISON en ACHAT ?
> 
> ```
> --      WHERE 
> --         EXISTS (
> --             SELECT 1
> --             FROM source_dvf d1
> --             WHERE d1.id_mutation = source_dvf.id_mutation AND d1.type_local IN ('Appartement', 'Maison')
> --         ) AND
> --         NOT EXISTS (
> --             SELECT 1
> --             FROM source_dvf d2
> --             WHERE d2.id_mutation = source_dvf.id_mutation AND d2.nature_mutation != 'Vente'
> --         )
> ```

Aucune raison "forte" :) 

Je me suis dit que les dépendances, les terrains et locaux commerciaux repondaient à une logique de modélisation différentes en terme de surface qui risquent d'être nulles d'ailleurs.

Par contre on peut imaginer dans la table finale une colonne contenant un array ou un dict avec toutes les composantes de la mutations. Par ex : 

id_mutation, valeur, surface, type_du_bien_principal, autres_biens
1                  ,  100000.         ,  MAISON.                        , [DEPENDANCE, DEPENDANCE, TERRAIN]

Tu penses à quoi ? N'hésite pas à implémenter directement ce qui te semble la bonne logique

**Comment by soufianekhoudmi** [comment]
Je t'ai ajouté comme contributeur exterieur comme ça tu peut clone le repo, créer une branche et des PR et enclencher la CI.
Impossible d'enclencher le CI via des forks (CI dépend de secrets d'une PG mise en place pour l'integreation)

**Comment by soufianekhoudmi** [comment]
Cool ! Merci Tim :) 

Les voici (je ne les ai pas ouvertes et pas testés)

dvf_2014 : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2014/full.csv
dvf_2015 : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2015/full.csv
dvf_2016 : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2016/full.csv
dvf_2017 : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2017/full.csv
dvf_2018 : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2018/full.csv


dvf_2014_dev : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2014_dev/34.csv
dvf_2015_dev : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2015_dev/34.csv
dvf_2016_dev : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2016_dev/34.csv
dvf_2017_dev : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2017_dev/31.csv
dvf_2018_dev : https://make-open-data-s3.fra1.digitaloceanspaces.com/dvf_2018_dev/34.csv

**Comment by aloysius-tim** [comment]
Hello Soufiane,
Merci pour les uploads - cependant AccessDenied sur dvf_2016_dev

**Comment by soufianekhoudmi** [comment]
Réparé :)

**Comment by aloysius-tim** [comment]
J'ai tenté de créer une table TMP par millésime pour les UNION par la suite afin d'opti un peu les traitements. Mais impossible de trouver une facon clean.
DBT ne permet pas d'avoir plusieurs queries ou creation de table dans le même fichier de ce que je comprend. Si tu as une reco à ce sujet (je connais tres tres peu DBT)

```
{% for millesime in millesimes %}
    {{
        config(
            materialized='table',
            schema='intermediaires',
            alias='dvf_' ~ millesime,
            post_hook=[
                "CREATE INDEX IF NOT EXISTS geopoint_index_{{ millesime }} ON {{ this }} USING GIST(geopoint);"
            ]
        )
    }}

    WITH source_dvf AS (
        SELECT *
        FROM {{ source('sources', millesime) }}
        LIMIT 100
    )
    SELECT
        id_mutation,
        valeur_fonciere,
        nature_mutation,
        longitude,
        latitude,
        surface_reelle_bati,
        ST_SetSRID(ST_MakePoint(latitude, longitude), 4326) as geopoint,
        code_postal,
        code_commune
    FROM
        source_dvf
{% endfor %}
```

**Comment by aloysius-tim** [comment]
Yes, je voulais justement éviter de devoir avoir un SQL par millésime

La CI est bien passée sur les datas de DEV. Je test sur ma DB avec tous les millésimes, a voir, c'est pas très opti niveau perf
